cognite_toolkit/_cdf_tk/utils/fileio/_readers.py

```diff
@@ -0,0 +1,55 @@
+from abc import ABC, abstractmethod
+from collections.abc import Iterator, Mapping
+from io import TextIOWrapper
+from pathlib import Path
+
+from cognite_toolkit._cdf_tk.exceptions import ToolkitValueError
+from cognite_toolkit._cdf_tk.utils._auxiliary import get_concrete_subclasses
+from cognite_toolkit._cdf_tk.utils.collection import humanize_collection
+from cognite_toolkit._cdf_tk.utils.useful_types import JsonVal
+
+from ._base import FileIO
+from ._compression import COMPRESSION_BY_SUFFIX, Compression
+
+
+class FileReader(FileIO, ABC):
+    def __init__(self, input_file: Path) -> None:
+        self.input_file = input_file
+
+    def read_chunks(self) -> Iterator[JsonVal]:
+        compression = Compression.from_filepath(self.input_file)
+        with compression.open("r") as file:
+            yield from self._read_chunks_from_file(file)
+
+    @abstractmethod
+    def _read_chunks_from_file(self, file: TextIOWrapper) -> Iterator[JsonVal]:
+        """Read chunks from the file."""
+        raise NotImplementedError("This method should be implemented in subclasses.")
+
+    @classmethod
+    def from_filepath(cls, filepath: Path) -> "FileReader":
```

Comment:
Can you add some tests for this method?

Reply:
Similarly to the from format, the same test also uses this when I introduce the concrete classes: https://github.com/cognitedata/toolkit/pull/1802/files#diff-8fca510d461329e4b2707e116ae1146f4dcb77a6536acf852abc92c290c71531R189

Reply:
Same applies here: https://github.com/cognitedata/toolkit/pull/1799#discussion_r2253680779